homeassistant/components/screenlogic/__init__.py

```diff
@@ -125,7 +133,15 @@ async def _async_update_data(self):
             async with self.api_lock:
                 await self.hass.async_add_executor_job(self.gateway.update)
         except ScreenLogicError as error:
-            raise UpdateFailed(error) from error
+            _LOGGER.warning("ScreenLogicError - attempting reconnect: %s", error)
+
+            connect_info = get_connect_info(self.hass, self.config_entry)
+            try:
+                self.gateway = ScreenLogicGateway(**connect_info)
+                self.gateway.update()
```

Comment:
`self.gateway.update()` should be wrapped with the `api_lock` and I believe called with `self.hass.async_add_executor_job`, same as line 133 & 134.
```suggestion
                async with self.api_lock:
                    await self.hass.async_add_executor_job(self.gateway.update)
```